### ISSUE TITLE
tdStyle - Anyway to call a function with the same information as dataFormat?

### ISSUE THREAD

**Issue by stevelacerda7** [AllenFang/react-bootstrap-table#1605]
I have a separate rule list that is set by the user, so I don't know what the style is beforehand. I am trying to style my td based on this object that I receive by the user. So, I'm trying something like:

```
<TableHeaderColumn` key={ id } dataField={ header } dataSort={ true } tdStyle={ this._statusStyler() }>{ header.toUpperCase() }</TableHeaderColumn>

_statusStyler(fieldValue, row, rowIdx, colIdx) {
	if (typeof(fieldValue) === "string" && this.ruleStatusListKeys.indexOf(fieldValue.toLowerCase() > -1)) {
		return {
			color: this.ruleStatusListKeys
		};
	}
	return {};
}
```

But, I need to pass some arguments just like dataFormat does, so that I know if that td meets the criteria to apply a rule, and which rule.

**Comment by stevelacerda7** [comment]
I'm working around this right now using:

```
<TableHeaderColumn key={ id } dataField={ header } dataSort={ true } dataFormat={ this._statusStyler }>{ header.toUpperCase() }</TableHeaderColumn>

// uses rule status setup by customer to decide how to format the columns
_statusStyler(cell, row, formatExtraData, rowIdx) {
	const currCell = this.state.ruleStatusList[cell];

	if (currCell) {
		return `<span class="${currCell['name']} ${currCell['blink'] ? 'blink' : ''}" style="color: ${currCell['color']}">${cell}</span> `; 
	}
	return cell;
}
```

But, it would be nice to have tdStyle accept an overload function with the same information as dataFormat.

**Comment by AllenFang** [comment]
I will handle this, thanks 👍

**Comment by AllenFang** [comment]
Fixed on `v4.1.0`, check https://github.com/AllenFang/react-bootstrap-table/blob/master/examples/js/column/column-style-table.js#L24
